terraform/a2ha-terraform/modules/opensearch/templates/provision.sh.tpl

```diff
@@ -89,3 +93,41 @@ if [ -e /hab/sup/default/specs/"$OPENSEARCHSIDECAR_PKG_NAME".spec ]; then
 else
   bash -c 'eval hab svc load ${opensearchsidecar_pkg_ident} ${opensearchsidecar_svc_load_args} --bind opensearch:"$OS_PKG_NAME".default --binding-mode=strict "$LOGCMD"'
 fi
+
+# 
+echo "${backup_config_s3}"
+
+SERVICE_UP_TIME=$(echo yes | hab svc status  | awk  '{print $5}' | tail -1)
+
+# Following loop will hold adding credentials in keystore until
+# services are up and ran for 30 seconds and gives enough
+# time to remove temp keystore file 'opensearch.keystore'
+
+# If you get following error, ===> `[: : integer expression expected`
+# try `until [[ "$SERVICE_UP_TIME" -gt 30 ]]` instead of `until [ "$SERVICE_UP_TIME" -gt 30 ]`
+if [ "${backup_config_s3}" == "true" ]; then
+
+max=15
+n=0
+until [ "$SERVICE_UP_TIME" -gt 30 and  $n -ge $max ]
+do
+  sleep 5
+  SERVICE_UP_TIME=$(echo yes | hab svc status  | awk  '{print $5}' | tail -1)
+  echo "No services are loaded"
+  echo "Sleeping for 5 seconds"
+  n=$((n+1))
+done
+
+# Adding aws access and secret keys once all services are up
+
+  echo "Setting up keystore"
+  echo yes | hab svc status
+  export OPENSEARCH_PATH_CONF="/hab/svc/automate-ha-opensearch/config"
+  echo $OPENSEARCH_PATH_CONF
+  echo "${access_key}" | hab pkg exec "$OS_ORIGIN_NAME/$OS_PKG_NAME" opensearch-keystore add --stdin --force s3.client.default.access_key
```

Comment:
what happens if the user is using IAM profiles to authenticate with AWS instead of IAM secret/keys?

Reply:
As of now, we are not supporting on-prem with AWS-managed services. Though we are planning to add some more config in upcoming sprints to support backup with IAM profile in AWS managed services